core/lib/basic_types/src/vm.rs

```diff
@@ -35,5 +35,9 @@ pub enum FastVmMode {
     /// Run only the new Vm.
     New,
     /// Run both the new and old VM and compare their outputs for each transaction execution.
+    /// The VM will panic on divergence.
     Shadow,
+    /// Run both the new and old VM and compare their outputs for each transaction execution.
+    /// The new VM will get dropped on divergence.
+    ShadowLenient,
```

Comment:
An alternative approach would be the dump handler to handle both dumps and errors:

```rust
pub struct VmDumpHandler(Arc<dyn Fn(VmDump, anyhow::Error) + Send + Sync>);
```

This allows to not introduce a new VM mode, and it would allow to implement `zksync_vlog::sentry` suggestion in a less intrusive way (IMO, making `multivm` depend on such a high-level crate is pushing it). OTOH, gluing handling of multiple things together looks slightly hacky. WDYT?

Reply:
Decided to do just that.